client/app/components/application.coffee

```diff
@@ -40,153 +30,58 @@ MessageActionCreator = require '../actions/message_action_creator'
     (based on:
         https://medium.com/react-tutorials/react-backbone-router-c00be0cf1592)
 ###
-module.exports = Application = React.createClass
+
+Application = React.createClass
     displayName: 'Application'
 
     mixins: [
-        StoreWatchMixin Stores
-        RouterMixin
         TooltipRefesherMixin
     ]
 
-    render: ->
-        # Shortcut
-        # TODO: Improve the way we display a loader when app isn't ready
-        layout = @props.router.current
-        return div null, t "app loading" unless layout?
-
-        disposition = LayoutStore.getDisposition()
-        isCompact   = LayoutStore.getListModeCompact()
-        fullscreen  = LayoutStore.isPreviewFullscreen()
-        previewSize = LayoutStore.getPreviewSize()
+    getDefaultProps: ->
+        ApplicationGetters.getProps 'application'
```

Comment:
@aenario : Are you still agree with this idea, because I AM :) It should be easier to implement, before adding a `redux` logic (or other stuff)